### ISSUE TITLE
Unclear condition "time → after ..." / Suggestion for adding at least "between"

### ISSUE THREAD

**Issue by just-Nob** [renyuneyun/Easer#262]
There is a condition "time" with the only attribute "after xx:yy (o'clock)".
The question is: What means "after"? Or, to say it more clear: How _long_ will "after" last?
For the rest of the day (i.e. until 12:00 p.m.)?
For one hour?
For the next 12 hours?

Supposed that it for the rest of the current day, another question came up to me:
Why there is only the condition "after"?
Why not also "before"?
And, last not least, why not "between"?

In my opinion, the latest one ("between") is the most important, because it's capable to cover "before" and "after" as well.

**Comment by just-Nob** [comment]
Ok - I've noticed your hint "before <--> combination of is_not and after". 
So it should be possible to build up a "between" condition by chaining two scripts.

_For example:_
If you want to turn off wifi between 11:00 a.m. and 01:00 p.m., the script chain for switching wifi off could look like this:
1. a) condition: time is not after 01:00 p.m.
b) parent: none
c) profile: none
2. a) condition: time is after 11:00 a.m.
b) parent: script "1."
c) profile: switch wifi off

A little more complicated, but should work, I guess (currently haven't checked it).

**Comment by renyuneyun** [comment]
Glad you saw that page on the (small) wiki :)

There used to be these things (after, before, between, etc) for every Event. Then that got removed (because I discovered that way is not right, because not everything has such properties).
The "After" text there is just to remind people that it is not "before".

**Comment by just-Nob** [comment]
Anyway, I'm still confused about the condition's duration: How long will "after" be true / valid?

**Comment by renyuneyun** [comment]
Yes, until the end of the day (because it's "time" without "date").

**Comment by scruloose** [comment]
@just-Nob: I found that pretty confusing as well, but here's what I've figured out:
* 'after' means "from hh:mm to midnight"
* a reversed 'after' is a 'before', which means "from midnight until hh:mm"
* for a "between hh:mm and xx:yy" range *within the same day*, you need to put the two in series, so it's only satisfied when the time is both after hh:mm and before xx:yy.
* for a "between hh:mm in the evening and aa:bb in the morning" range, you need to put them in parallel (ie two scripts with a common parent, both calling the same Profile), so it's satisfied if either "from hh:mm to midnight" or "from midnight to aa:bb" is true.

All of these will automatically repeat every day, unless you explicitly chain a date condition in there too.

@renyuneyun: In terms of the long-term plan of having the UI translate things from the backend logic to present the user with more intuitive input options, I think this issue would be an excellent candidate. If we're hoping to win people over from the proprietary solution (Tasker), the ability to set conditions like "from 9pm to 7am every day" will be accessible to a much larger cross-section of Android users than the current procedure.

**Comment by just-Nob** [comment]
Hi @scruloose,

thanks for the explanation of the workaround, but wasn't that exactly the same idea that I described in the second post of this issue (https://github.com/renyuneyun/Easer/issues/262#issuecomment-526362387 ), or is there any difference?

**Comment by DoctorSubtilis** [comment]
> Ok - I've noticed your hint "before <--> combination of is_not and after".
> So it should be possible to build up a "between" condition by chaining two scripts.
> 
> _For example:_
> If you want to turn off wifi between 11:00 a.m. and 01:00 p.m., the script chain for switching wifi off could look like this:
> 
>     1. a) condition: time is not after 01:00 p.m.
>        b) parent: none
>        c) profile: none
> 
>     2. a) condition: time is after 11:00 a.m.
>        b) parent: script "1."
>        c) profile: switch wifi off
> 
> 
> A little more complicated, but should work, I guess (currently haven't checked it).

ops: this trick didn't work for me.  
But why, @renyuneyun, you don't plan to let us choose a condition *between* (a start/an end)?  
Any way, thank you for your beatiful app, long life to open source!
